services/movie_session.py

```diff
@@ -0,0 +1,58 @@
+from django.core.exceptions import ObjectDoesNotExist
+from db.models import MovieSession, CinemaHall, Movie
+from datetime import date, datetime
+
+
+def create_movie_session(
+        movie_show_time: datetime,
+        movie_id: int,
+        cinema_hall_id: int,
+) -> MovieSession:
+    return MovieSession.objects.create(
+        show_time=movie_show_time,
+        movie_id=movie_id,
+        cinema_hall_id=cinema_hall_id,
+    )
+
+
+def get_movies_sessions(session_date: str = "") -> MovieSession | None:
+    if session_date:
+        year, month, day = session_date.split("-")
+        return MovieSession.objects.filter(
+            show_time__date=date(int(year), int(month), int(day)),
+        )
+    return MovieSession.objects.all()
+
+
+def get_movie_session_by_id(movie_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=movie_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: datetime = None,
+        movie_id: int = None,
+        cinema_hall_id: int = None
+) -> None:
+    try:
+        movie_session = MovieSession.objects.get(id=session_id)
+    except ObjectDoesNotExist:
+        print(f"Movie Session with ID {session_id} does not exist")

```

Comment:
This is not so good pattern - if record doesn't exist, warning should be raised and execution must be stopped, if you return None instead of a Movie it may lead to unpredictable behavior in other parts of the program